src/javascript/javascript-parser.ts

```diff
@@ -92,45 +102,56 @@ export function parseJs(
     contents: string,
     file: string,
     locationOffset?: LocationOffset,
-    warningCode?: string): ParseResult {
+    warningCode?: string,
+    sourceType?: SourceType): ParseResult {
   if (!warningCode) {
     warningCode = 'parse-error';
   }
-  const options = Object.assign(
-      {sourceType: 'script' as ('script' | 'module')}, baseParseOptions);
+
+  let program: estree.Program;
+
   try {
+    if (!sourceType) {
+      try {
+        sourceType = 'script';
+        program = espree.parse(
+            contents,
+            Object.assign(
```

Comment:
```js
espree.parse(contents, {sourceType: 'script' as SourceType, ...baseParseOptions});
```

Reply:
Nice!  I didn't realize `...` was for more than just array-type splatting.